ui/index.js

```diff
@@ -34,7 +34,8 @@ function startApp (metamaskState, accountManager, opts) {
   })
 
   // if unconfirmed txs, start on txConf page
-  if (Object.keys(metamaskState.unconfTxs || {}).length) {
+  var unconfirmedTxsAll = txHelper(metamaskState.unconfTxs, metamaskState.unconfMsgs, metamaskState.network)
+  if (unconfirmedTxsAll > 0) {
```

Comment:
I guess this actually works correctly as written, but I wasn't sure upon reading it, so I'd incline towards doing it the more obvious way.